blockchain/event_filter.go

```diff
@@ -144,7 +147,7 @@ func (e *EventFilter) Events(cToken *ContinuationToken, chunkSize uint64) ([]*Fi
 				return nil, nil, err
 			}
 		} else {
-			header = pending.Block.Header
+			header = pending.Header
```

Comment:
The nil check is done above on line 125. Since we are setting the limit on the block number in the if statement on line 121, it is not possible for this else case to execute if pending is nil.

Reply:
I see, now latest+1 makes sense. thanks